framework/include/times/Times.h

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose includes
+#include "GeneralReporter.h"
+
+/**
+ * Times objects are under the hood Reporters, but limited to a vector of Real
+ */
+class Times : public GeneralReporter
+{
+public:
+  static InputParameters validParams();
+  Times(const InputParameters & parameters);
+  virtual ~Times() = default;
+
+  /// Getter for the full times vector
+  const std::vector<Real> & getTimes() const;
+
+  /// Getter for a set of the full times
+  const std::set<Real> getUniqueTimes() const
+  {
+    return std::set<Real>(_times.begin(), _times.end());
+  };
+
+  /// Get the current time
+  Real getCurrentTime() const { return _fe_problem.time(); }
+
+  /// Getter for a single time at a known index
+  Real getTimeAtIndex(unsigned int index) const;
+
+  /// Find the previous time in the times vector for a given time
+  /// If current_time is also in the times vector within numerical precision, will return the previous value
+  Real getPreviousTime(const Real current_time) const;
```

Comment:
```suggestion
  /// If the times are not sorted or unique, this will return the time right before the first time found above the current_time
  Real getPreviousTime(const Real current_time) const;
```